ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/EngineCapabilitiesMonitor.java

```diff
@@ -0,0 +1,93 @@
+ */
+
+package tech.pegasys.teku.ethereum.executionlayer;
+
+import com.google.common.base.Suppliers;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.ethereum.events.SlotEventsChannel;
+import tech.pegasys.teku.ethereum.executionclient.ExecutionEngineClient;
+import tech.pegasys.teku.ethereum.executionclient.response.ResponseUnwrapper;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.logging.EventLogger;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+
+public class EngineCapabilitiesMonitor implements SlotEventsChannel {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final int EPOCHS_PER_MONITORING = 10;
+  private static final UInt64 SLOT_IN_THE_EPOCH_TO_RUN_MONITORING = UInt64.valueOf(3);
+
+  private final AtomicReference<UInt64> lastRunEpoch = new AtomicReference<>();
+
+  private final Spec spec;
+  private final EventLogger eventLogger;
+  private final Supplier<List<String>> capabilitiesSupplier;
+  private final ExecutionEngineClient executionEngineClient;
+
+  public EngineCapabilitiesMonitor(
+      final Spec spec,
+      final EventLogger eventLogger,
+      final EngineJsonRpcMethodsResolver engineMethodsResolver,
+      final ExecutionEngineClient executionEngineClient) {
+    this.spec = spec;
+    this.eventLogger = eventLogger;
+    this.capabilitiesSupplier = Suppliers.memoize(engineMethodsResolver::getCapabilities);
+    this.executionEngineClient = executionEngineClient;
+  }
+
+  @Override
+  public void onSlot(final UInt64 slot) {
+    final UInt64 currentEpoch = spec.computeEpochAtSlot(slot);
+    if (epochIsApplicable(currentEpoch) && slotIsApplicable(slot)) {
+      monitor()
+          .handleException(ex -> LOG.error("Exception exchanging Engine API capabilities", ex))
+          .always(() -> lastRunEpoch.set(currentEpoch));
+    }
+  }
+
+  private boolean epochIsApplicable(final UInt64 epoch) {
+    return lastRunEpoch.get() == null
+        || epoch.minusMinZero(lastRunEpoch.get()).isGreaterThanOrEqualTo(EPOCHS_PER_MONITORING);
+  }
+
+  private boolean slotIsApplicable(final UInt64 slot) {
+    return slot.mod(spec.getSlotsPerEpoch(slot))
+        .equals(SLOT_IN_THE_EPOCH_TO_RUN_MONITORING.minusMinZero(1));
+  }
+
+  private SafeFuture<Void> monitor() {
+    final List<String> capabilities = capabilitiesSupplier.get();
+    return executionEngineClient
+        .exchangeCapabilities(capabilities)
+        .thenApply(ResponseUnwrapper::unwrapExecutionClientResponseOrThrow)
+        .thenAccept(
+            engineCapabilities -> {
+              LOG.debug("Engine API capabilities: " + engineCapabilities);
```

Comment:
Might be worth specifying that these are the method returned from the remote peer. Something like:

```
LOG.debug("Engine API capabilities response: " + engineCapabilities);
```